tests/upgrades.rs

```diff
@@ -0,0 +1,64 @@
+use crate::common::TestConfig;
+use anyhow::Result;
+use espresso_types::{FeeVersion, MarketplaceVersion, V0_1};
+use futures::{stream, StreamExt};
+use vbs::version::StaticVersionType;
+
+const SEQUENCER_BLOCKS_TIMEOUT: u64 = 120;
+
+#[async_std::test]
+async fn test_upgrade() -> Result<()> {
+    dotenvy::dotenv()?;
+
+    let testing = TestConfig::new().await.unwrap();
+
+    let versions = if testing.sequencer_version >= 3 {
+        (FeeVersion::version(), MarketplaceVersion::version())
+    } else {
+        (V0_1::version(), FeeVersion::version())
+    };
+
+    println!("Waiting on readiness");
+    let _ = testing.readiness().await?;
+
+    let initial = testing.test_state().await;
+    println!("Initial State:{}", initial);
+
+    let clients = testing.sequencer_clients;
+
+    // Test is limited to those sequencers with correct modules
+    // enabled. It would be less fragile if we could discover them.
+    let subscriptions = vec![
+        clients[0].subscribe_headers(0).await?,
+        clients[1].subscribe_headers(0).await?,
+    ];
+    let subscriptions_size = subscriptions.len();
+    let mut streams = stream::iter(subscriptions)
+        .flatten_unordered(usize::try_from(SEQUENCER_BLOCKS_TIMEOUT).ok());
```

Comment:
why are we passing `SEQUENCER_BLOCKS_TIMEOUT` here

Reply:
It provides a limit, but I had misread the docs as to what it was limiting. 

https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.flatten_unordered

Reply:
eb3cd72eea26ab5f2218fdb53f68f6b24ff882a7